socketshark/subscription.py

```diff
@@ -116,6 +128,44 @@ def _should_deliver_message_order(self, data):
             self.order_state[key] = order
         return True
 
+    def _should_deliver_message_throttle(self, data):
+        """
+        Returns whether to deliver the given message based on throttling.
+        """
+        if '_throttle' not in data:
+            return True
+
+        try:
+            throttle = float(data['_throttle'])
+        except (TypeError, ValueError):
+            return False  # Don't deliver messages with invalid throttle.
+
+        key = data.get('_throttle_key')
+        last_throttle = self.throttle_state.get(key)
+        now = time.time()
+        if last_throttle:
+            ts_last_msg_sent, pending_msg, task = last_throttle
+            if task:  # We'll update the message and let the task send it.
+                self.throttle_state[key] = (ts_last_msg_sent, data, task)
```

Comment:
Do you want trace level logging here at least in the beginning to be able to confirm this thing is working?

Reply:
I added a few more log statements though I think this particular one should be fine since we're already getting a `message filtered` log.